### ISSUE TITLE
lost completely

### ISSUE THREAD

**Issue by commentsonly** [bruxy70/Home-Assistant-ESPHome-Weather-Station#10]
watched some of your videos aswell as trying to understand the code.
Even no problem with the GUI editor from nextion.
But what's missing is another 5min to your videos about the HOW TO integrate things with HA.
A simply .. this should go there .. isn't enough information.
Same applies to the HMI file. I'm lost guessing if this must be warped into the display onto an SD bumped into the displays SD card slot or simply should sit inside a HA ESPhome project.
Since so, thanks for spreading the word and videos about that things should work as expexted but it's still a miracle to me about how to get these things together. Looks to me as if I did understand if ever 10% of all steps one has to do.
Perhaps you could do another video showing the step by step about where to place each single file from here into which target, let if be ESPhome or HA.

**Comment by bruxy70** [comment]
This repository is for [this video](https://youtu.be/_u1sOCfyghY). Sorry, can't do better than that.

**Comment by commentsonly** [comment]
Don't get me wrong I appreciated exactly that video a lot. It was the first one I stumbled across.
As said earlier, what I do miss is likely the steps poeple more familar with HA won't need a second to think about.
but I'm a step further now ... from the ESPhome docs I see that the HMI has to go onto the SD inserted into the display.
Might take some some weeks until I found out everything all others already know about.
And finally ..... thanks for the damn fast response, I didn't expect you acting that fast.

**Comment by commentsonly** [comment]
getting closer to a solution, just wanted to let you know. :-) issue in the begining, my old USB=TTL=adapter was the problem. While being able to use if flashing gosund plugs and such like, it never made it. Not to mention the nextion refused most of my microSD cards. With a newly bough USB=TTL=adapter it works to flash the device.
And I'm also also able to blow values into the textboxes.
But I failed to work out about how to blow sensor values onto the device. I got confused with all these terms, id, entitly_id, and so on. I assumed the entitly_id must be the one from HA but tuning on logging does show these remain empty which explains why the display textboxes show 'nan' all the time.
I thought that I could simply make use of what HA lists from within settings entities entity-id. Am I wrong here?

**Comment by bruxy70** [comment]
Hi. I actually never tried to use the SD card with the Nextion. I think what you ask is described in the Esphome documentation. It worked for me quite smoothly, if that doe snot work for you, I suggest asking this to the Esphome support, I am not really expert in this. Sorry.

**Comment by commentsonly** [comment]
as said I bypassed this SD card issue by now using a new USB-TTL-adapter.
All I got stuck is that I can't get any sensor values sent to the nextion.
in short
it.set_component_text("objectname","1234"); does work while using an id linked to an entity_id availalbe within homeassistant simply results in '' (nothing).
Means it's surely not an issue between the nextion and the esp connected to it, but more something to do with
"  - platform homeassistant" which I never came across right now. I was assuming that if I got an esp linked to Ha with a BME280 sensor ending up with a temperaturesensor resulting in esp8266_outside_temp ..... it should be fine to use that entity_id with the sensor: - platform: homeassistant
  id: esp8266_outside_temp
  entity_id: sensor.esp8266_outside_temp
but I might have been wrong here. The sensor does show values if looking at developper in HA were all the current values are shown besides all entities, but when logging what happens in esphome with logging turned on the result is simply ''.
And if nothing reaches the esp connected to the nextion the nextion won't display anything but 'nan'.

**Comment by commentsonly** [comment]
Oh my god ...... got it running, my fault.
I mixed up "Home Assistant" with "Home Automation", means I forgot about 1 small step for the noob but a big step to make you guideline 100% working against 100% failing.
If one flashes an ESP from with ESPhome fired up from within HA there's automation in terms of integrating the new ESP into HA. And since so the new ESP won't see any HA entities at all, resulting in '' with then leads to 'nan'. That simple.
Found an article that indicated I wasn't the first who suffered from that issue.
Perhaps a simple 1-liner not at the ned "don't forget about..." would have made things lots easier.

Nevertheless ... great work from your side with the descriptions.
